scripts/build.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python
+
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Assuming pre-requisites are in place (please run `scripts/install_preqreqs.py` once on the system), 
+this script will build the firebase cpp sdk
+
+It does the following,
+- install platform specific cpp dependencies via vcpkg
+- cmake configure
+- cmake build
+
+Usage examples:
+# Build all targets with x64 architecture, Release config and 8 cpu cores fo
+python build.py --arch x64 --config Release -j 8
+
+# Build all targets with default options and also build unit tests
+python build.py --build_tests --arch x64
+
+# Build only firebase_app and firebase_auth with default options and also build unit tests
+python build.py --build_tests --arch x64 --targets firebase_app firebase_auth
+
+"""
+
+import subprocess
+import os
+import sys
+import argparse
```

Comment:
Since we're installing absl-py anyway and we're trying to stick to the google python style guide wherever possible, WDYT about using absl.flags instead of argparse?

Reply:
If were assuming access to absl-py, that also gives us access to absl's logging module. Instead of print, we can use logging.info, logging.warning and logging.error which will label the message with a timestamp, line number, and log level (I, W, E). Having that extra info can be quite helpful in a number of situations.

For consistency, we probably should decide whether tools in this repo should make use of absl-py or not.

Reply:
I am on the fence about this. For simple scripts like this, we should try to keep them as lean as possible. Python has standard library modules for command line argument parsing and logging (not the best but is workable and can set formatting and log levels). Also, since we are trying to create something very close to a shell script, we probably wont use logging much because it adds verbosity. Google python style guide doesn't enforce its usage and taking a quick glance at firebase-ios-sdk, they are using standard logging and arparse modules in their python scripts too.

Reply:
I don't think absl.flags vs argparse makes a huge difference, but I think there is value in picking one and using that for all of our scripts to minimize the difficulty in contributing/maintaining the repo. (Mind you I'm guilty of violating this - my PR has absl.flags for almost everything, but then argparse for the restore_secrets script. I plan to rewrite one of them once we decide on which way to go with this).  

I'm not sure I follow the shell script argument though. Adding a bit of extra verbosity for useful diagnostic information seems like a good tradeoff. If nothing else, it seems very valuable to know how long each step is taking, which the timestamps give us. 

I don't think it's a huge deal either way, but I think it makes sense to use absl-py, for both flags and logging.

Reply:
Will be creating another PR incase we decide to go absl.